frame/benchmarking/src/lib.rs

```diff
@@ -1050,7 +1057,18 @@ macro_rules! add_benchmark {
 							*repeat,
 							whitelist,
 							*verify,
-						)?,
+						).map_err(|e| { 
+							$crate::join_u8_sequences(
```

Comment:
Why not use `format!`?

Reply:
I think `format!` is not available in a `no_std` environment. Is it?

Reply:
It is ;) `sp_std::alloc::format!`, but :shushing_face:

Reply:
oh thanks. that would make things a lot easier. 
so if I replace the `join_u8_sequences(...)` call with something like `sp_std::alloc::format!("blabla {}", 1u32).as_bytes().to_vec()`, recompiling fails with 
```
   Compiling node-template-runtime v2.0.0 (/Users/simon/dev/substrate/bin/node-template/runtime)
error[E0433]: failed to resolve: could not find `format` in `alloc`
   --> bin/node-template/runtime/src/lib.rs:471:4
    |
471 |             add_benchmark!(params, batches, pallet_timestamp, Timestamp);
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ could not find `format` in `alloc`
    |
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: could not find `format` in `alloc`
   --> bin/node-template/runtime/src/lib.rs:470:4
    |
470 |             add_benchmark!(params, batches, pallet_balances, Balances);
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ could not find `format` in `alloc`
    |
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: could not find `format` in `alloc`
   --> bin/node-template/runtime/src/lib.rs:469:4
    |
469 |             add_benchmark!(params, batches, frame_system, SystemBench::<Runtime>);
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ could not find `format` in `alloc`
    |
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 3 previous errors
```
what am I doing wrong?

Reply:
I mean I see that `without_std.rs` does `public extern crate alloc;` and that `format!` is in there. but I can't seem to get it to work here.

Reply:
it is hard so say why without the code, did you use sp_std::alloc or just `alloc` ?

Reply:
Ahh that error is probably from the native build? For native we don't export the alloc crate.

Reply:
ok. so `format!` is not available then?

Reply:
On native `format!` is provided by the rustc prelude, meaning it is "just" available.

So you need to use some cfgs around this call.

Reply:
Aah, thanks @bkchr and @thiolliere for enlightening me. I wasn't aware of the fact that it was compiled once _with_ and once _without_ `std` features. Now it makes a lot more sense to me.